modules/openapi-generator-gradle-plugin/build.gradle

```diff
@@ -1,84 +1,56 @@
-buildscript {
-    ext.kotlin_version = '1.3.30'
-    repositories {
-        mavenLocal()
-        maven { url "https://repo1.maven.org/maven2" }
-        maven {
-            url "https://plugins.gradle.org/m2/"
-        }
-        maven {
-            url "https://oss.sonatype.org/content/repositories/releases/"
-        }
-        maven {
-            url "https://oss.sonatype.org/content/repositories/snapshots/"
-        }
-    }
-    dependencies {
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-        classpath "gradle.plugin.org.gradle.kotlin:gradle-kotlin-dsl-plugins:1.1.3"
-        classpath "com.gradle.publish:plugin-publish-plugin:0.11.0"
-        classpath "io.codearte.gradle.nexus:gradle-nexus-staging-plugin:0.20.0"
-        classpath "de.marcphilipp.gradle:nexus-publish-plugin:0.2.0"
-    }
-}
+import io.github.gradlenexus.publishplugin.CloseNexusStagingRepository
+import io.github.gradlenexus.publishplugin.ReleaseNexusStagingRepository
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    id("com.gradle.plugin-publish") version "1.0.0"
+    id("io.github.gradle-nexus.publish-plugin") version "1.1.0"
+    id("java-gradle-plugin")
+    id("maven-publish")
+    id("org.gradle.kotlin.kotlin-dsl") version "2.4.1"
+    id("org.jetbrains.kotlin.jvm") version "1.7.10"
+    id("signing")
+}
+
+group = "org.openapitools"
+version = "$openApiGeneratorVersion"
+ext.isReleaseVersion = !version.endsWith("SNAPSHOT")
 
-group 'org.openapitools'
-// Shared OpenAPI Generator version be passed via command line arg as -PopenApiGeneratorVersion=VERSION
-version "$openApiGeneratorVersion"
 description = """
-This plugin supports common functionality found in Open API Generator CLI as a gradle plugin.
+This plugin supports common functionality found in Open API Generator CLI as a Gradle plugin.
 
 This gives you the ability to generate client SDKs, documentation, new generators, and to validate Open API 2.0 and 3.x
 specifications as part of your build. Other tasks are available as command line tasks.
 """
-ext.isReleaseVersion = !version.endsWith("SNAPSHOT")
-
-apply plugin: 'com.gradle.plugin-publish'
-apply plugin: 'java-gradle-plugin'
-apply plugin: 'signing'
-apply plugin: 'kotlin'
-apply plugin: "org.gradle.kotlin.kotlin-dsl"
-apply plugin: 'io.codearte.nexus-staging'
-apply plugin: "de.marcphilipp.nexus-publish"
 
-sourceCompatibility = 1.8
-targetCompatibility = 1.8
+java {
+    withSourcesJar()
+    withJavadocJar()
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
+}
 
 repositories {
-    mavenLocal()
```

Comment:
@erichaagdev I think we will need to add this back. Please take a look at the build failure in this PR: https://github.com/OpenAPITools/openapi-generator/pull/14172. 

After adding back this, I can no longer repeat the issue locally via "mvn clean install"

Reply:
Ah, I do think it should be there. Good catch. Do you want me to add it back or will you add it in your PR?

Reply:
However the PR failure I am looking at seems to be related to connectivity, not about resolving dependencies from Maven local.

https://github.com/OpenAPITools/openapi-generator/actions/runs/3620526865/jobs/6102851542#step:5:114

Reply:
```
> Task :publishOpenApiGeneratorPluginMarkerMavenPublicationToMavenLocal
> Task :generatePomFileForPluginMavenPublication

> Task :compileKotlin FAILED
e: /home/runner/work/openapi-generator/openapi-generator/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/GenerateTask.kt: (770, 34): Unresolved reference: addOpenAPINormalizer
```

We added a new method `addOpenAPINormalizer` and it's only available in the local maven cache store. 

I already filed https://github.com/OpenAPITools/openapi-generator/pull/14265

Reply:
UPDATE: merged #14265